inbm/dispatcher-agent/tests/unit/schedule/test_sqlite_manager.py

```diff
@@ -53,22 +56,28 @@ def test_raise_exception_when_create_repeated_schedule_with_no_manifests(db_conn
 
 def test_create_simple_schedule(db_connection: SqliteManager):
     db_connection.clear_database()
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Consider using parameterized tests for testing various manifest scenarios.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
@pytest.mark.parametrize("start_time, end_time, manifests, expected_exception", [
    (datetime.strptime("2024-01-01T00:00:00", "%Y-%m-%dT%H:%M:%S"), datetime.strptime("2024-01-02T00:00:00", "%Y-%m-%dT%H:%M:%S"), ["MANIFEST1", "MANIFEST2"], None),
    (None, datetime.strptime("2024-01-01T02:00:00", "%Y-%m-%dT%H:%M:%S"), ["MANIFEST1"], DispatcherException),
    # Add more test cases here
])
def test_create_schedule_with_various_manifests(db_connection: SqliteManager, start_time, end_time, manifests, expected_exception):
    ss1 = SingleSchedule(request_id="4324a262-b7d1-46a7-b8cc-84d934c3983f",
                         job_id="swupd-939fe48c-32da-40eb-a00f-acfdb43a5d6d",
                         start_time=start_time,
                         end_time=end_time,
                         manifests=manifests)
    if expected_exception:
        with pytest.raises(expected_exception) as excinfo:
            db_connection.create_schedule(ss1)
    else:
        db_connection.create_schedule(ss1)
        # Additional assertions can be added here
```

Reply:
Added for single schedule.